types/next/index.d.ts

```diff
@@ -131,8 +131,16 @@ declare namespace next {
      * @template C Context passed to getInitialProps.
      */
     interface NextStaticLifecycle<IP, C> {
-        getInitialProps?: (ctx: C) => Promise<IP> | IP;
+        getInitialProps?: GetInitialProps<IP, C>;
     }
+
+    /**
+     * Next.js getInitialProps type.
+     *
+     * @template IP Initial props returned from getInitialProps and passed to the component.
+     * @template C Context passed to getInitialProps.
+     */
+    type GetInitialProps<IP, C> = (ctx: C) => Promise<IP> | IP;
```

Comment:
I think this `C` should be defaulted to `NextContext`. I have a plan to extend all context objects from this base interface and it makes sense to default to the base interface.

Reply:
I think this would fail on App.getInitialProps (_app.tsx)

Reply:
@ClassicDarkChocolate You might be able to fix it by having `NextAppContext` extend `NextContext`, which is what I've been planning to do at some point.

Reply:
@resir014 I think they're not compatible and have different structure.

Reply:
I see. Let's keep it this way for now.